pkg/cloudprovider/providers/aws/aws.go

```diff
@@ -1156,6 +1163,33 @@ func (self *awsDisk) describeVolume() (*ec2.Volume, error) {
 	return volumes[0], nil
 }
 
+// getAttachmentStatus returns current attachment status of the volume
+func (self *awsDisk) getAttachmentStatus() (string, error) {
+	info, err := self.describeVolume()
+	if err != nil {
+		return "", err
+	}
+	if len(info.Attachments) > 1 {
+		glog.Warningf("Found multiple attachments for volume: %v", info)
+	}
+	attachmentStatus := ""
+	for _, attachment := range info.Attachments {
+		if attachmentStatus != "" {
+			glog.Warning("Found multiple attachments: ", info)
+		}
+		if attachment.State != nil {
+			attachmentStatus = *attachment.State
+		} else {
+			// Shouldn't happen, but don't panic...
+			glog.Warning("Ignoring nil attachment state: ", attachment)
+		}
+	}
+	if attachmentStatus == "" {
+		attachmentStatus = "detached"
```

Comment:
I agree not to make it an enum, given EBS has not made it an enum.  I believe there are likely to be other states that happen during EBS outages, or if your EBS volume is corrupted...